config/src/main/java/com/typesafe/config/impl/ConfigNodeObject.java

```diff
@@ -12,32 +12,83 @@
         super(children);
     }
 
-    protected ConfigNodeObject changeValueOnPath(Path desiredPath, AbstractConfigNodeValue value) {
+    public boolean hasValue(Path desiredPath) {
+        for (AbstractConfigNode node : children) {
+            if (node instanceof ConfigNodeField) {
```

Comment:
if you put `ConfigNodeField field = (ConfigNodeField) node` here and then use `field` below, then you could avoid some of the cast noise in the code below.